core/src/main/java/org/elasticsearch/transport/netty/MessageChannelHandler.java

```diff
@@ -226,14 +226,17 @@ protected String handleRequest(Channel channel, Marker marker, StreamInput buffe
         transportServiceAdapter.onRequestReceived(requestId, action);
         NettyTransportChannel transportChannel = null;
         try {
-            transport.inFlightRequestsBreaker().addEstimateBytesAndMaybeBreak(messageLengthBytes, "<transport_request>");
-            transportChannel = new NettyTransportChannel(transport, transportServiceAdapter, action, channel,
-                requestId, version, profileName, messageLengthBytes);
-
             final RequestHandlerRegistry reg = transportServiceAdapter.getRequestHandler(action);
             if (reg == null) {
                 throw new ActionNotFoundTransportException(action);
             }
+            long reservedBytes = 0;
+            if (reg.isCheckSizeLimit()) {
```

Comment:
Am I reading this right? We are not counting bytes from requests that are not subject to the limits in the sum of bytes of all outstanding requests? Is that we want? Or do we want to include the bytes but just not subject the request to the threshold check?